packages/api/src/controllers/animalBatchController.js

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalBatchModel from '../models/animalBatchModel.js';
+import baseController from './baseController.js';
+import DefaultAnimalBreedModel from '../models/defaultAnimalBreedModel.js';
+import CustomAnimalBreedModel from '../models/customAnimalBreedModel.js';
+import CustomAnimalTypeModel from '../models/customAnimalTypeModel.js';
+import { handleObjectionError } from '../util/errorCodes.js';
+
+const animalBatchController = {
+  getFarmAnimalBatches() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalBatchModel.query()
+          .where({ farm_id })
+          .whereNotDeleted()
+          .withGraphFetched('sex_detail');
+        return res.status(200).send(rows);
+      } catch (error) {
+        console.error(error);
+        return res.status(500).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addAnimalBatches() {
```

Comment:
I believe this is covered by the new errorCodes handler so it will be inconsistent yes. That one should now look like this: 

The different error codes I think you are referring to is the 403 forbidden. The original addAnimals check also checks for the existence of the type so I thought it was different. I could probably have simplified the if with a null check on the object maybe.

AddAnmials()
```
if (!customType || customType.farm_id !== farm_id) {
              await trx.rollback();
              return res.status(400).send('custom_type_id has invalid value');
            }
```

AddAnimalsBatch()
```
if (customType && customType.farm_id !== farm_id) {
              await trx.rollback();
              return res.status(403).send('Forbidden custom type does not belong to this farm');
            }
```

Reply:
Yeah, I don't like the subttle differences 😅 
Maybe it's fine for now, but if I looked back in a few years and compare the two, I would wonder why they are doing the same thing differently. (So, I want the new errorCodes handler to be used in `addAnimals` for example)
But everyone else is fine, I'm fine too.

Reply:
@SayakaOno I definitely agree with that perspective -- is it okay if I create a new jira ticket and try to update all the other animal routes?

Reply:
That sounds great, thank you!